Modules/Relationships/src/Relationships.Infrastructure/Persistence/Database/EntityTypeConfigurations/RelationshipTemplateEntityTypeConfiguration.cs

```diff
@@ -11,5 +11,12 @@ public void Configure(EntityTypeBuilder<RelationshipTemplate> builder)
         builder.HasIndex(x => x.CreatedBy);
         builder.HasIndex(x => x.DeletedAt);
         builder.HasIndex(x => x.ExpiresAt);
+
+        builder
```

Comment:
I don't understand what you're doing here. All you should have done was removing the questionmark from the `Relationship.RelationshipTemplate` and `Relationship.RelationshipTemplateId` property.

Reply:
When I created migrations with only that change, the migrations had a OnDeleteCascade for the RelationshipTemplate. This was not the behavior we had before so I tried changing it so the deletion gets Restricted when there are Relationships using the current RelationshipTemplate.

To implement that in the Configuration, I had to explicitly write out the relation between the Relationship Template table and the Relationship table.

Reply:
Ohh, I see. In that case: well done! :)